src/bun.js/api/server/RequestContext.zig

```diff
@@ -1664,13 +1660,13 @@ pub fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool,
 
             if (req.response_ptr) |resp| {
                 assert(req.server != null);
+                const globalThis = req.server.?.globalThis;
                 const bodyValue = resp.getBodyValue();
-                if (bodyValue.* == .Locked) {
-                    const global = bodyValue.Locked.global;
-                    if (bodyValue.Locked.readable.get(global)) |stream| {
-                        stream.done(global);
-                    }
-                    bodyValue.Locked.readable.deinit();
+                if (resp.getBodyReadableStream(globalThis)) |stream| {
+                    stream.value.ensureStillAlive();
+                    resp.detachReadableStream(globalThis);
+
+                    stream.done(globalThis);
```

Comment:
This subtly changes the semantics. now we only mark it as used if there is a ReadableStream attached. Previously, we unconditionally marked it as used. That is a bug.